File: cloud-code/UseInventoryItem.js
// UseInventoryItem Cloud Code Function
const { EconomyApi } = require("@unity-services/economy-1.0");

module.exports = async ({ params, context, logger }) => {
  try {
    const { itemId, quantity = 1 } = params;
    if (!itemId) throw new Error("Missing required parameter: itemId");
    if (quantity <= 0) throw new Error("Quantity must be positive");
    
    const inventory = await EconomyApi.getInventoryItems();
    const item = inventory.find(i => i.id === itemId);
    
    if (!item || item.quantity < quantity) throw new Error("Insufficient inventory items");
    
    await EconomyApi.useInventoryItem({ itemId, quantity });
    logger.info(`Used ${quantity} ${itemId} from player inventory`);
    
    return { success: true, itemId, quantity, remainingQuantity: item.quantity - quantity };
  } catch (error) {
    logger.error(`UseInventoryItem failed: ${error.message}`);
    throw error;
  }
};

File: cloud-code-csharp/src/UseInventoryItem.cs
using Unity.Services.CloudCode;
using Unity.Services.Economy;
using Unity.Services.Core;
using System;
using System.Threading.Tasks;
using System.Linq;

[CloudCodeFunction("UseInventoryItem")]
public class UseInventoryItem : CloudCodeFunction
{
    public async Task<object> ExecuteAsync(CloudCodeRequest request)
    {
        try
        {
            // Parse parameters
            var itemId = request.GetParameter<string>("itemId");
            var quantity = request.GetParameter<int>("quantity", 1);

            // Validate parameters
            if (string.IsNullOrEmpty(itemId))
                throw new ArgumentException("itemId is required");
            
            if (quantity <= 0)
                throw new ArgumentException("quantity must be positive");

            // Get current inventory
            var inventory = await EconomyService.Instance.Inventory.GetInventoryItemsAsync();
            var item = inventory.FirstOrDefault(i => i.InventoryItemId == itemId);

            if (item == null || item.Amount < quantity)
                throw new InvalidOperationException("Insufficient inventory items");

            // Use inventory item
            await EconomyService.Instance.Inventory.UseInventoryItemAsync(itemId, quantity);

            // Log success
            Logger.LogInfo($"Used {quantity} {itemId} from player inventory");

            return new
            {
                success = true,
                itemId = itemId,
                quantity = quantity,
                remainingQuantity = item.Amount - quantity
            };
        }
        catch (Exception ex)
        {
            Logger.LogError($"UseInventoryItem failed: {ex.Message}");
            throw;
        }
    }
}